tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,269 @@
+from typing import Dict, List, Optional, ClassVar, Set
+
+import pytest
+from unittest import mock
+
+from great_expectations.core.domain import Domain
+from great_expectations.core.metric_domain_types import MetricDomainTypes
+from great_expectations.rule_based_profiler.config import ParameterBuilderConfig
+from great_expectations.rule_based_profiler.parameter_builder import (
+    ParameterBuilder,
+)
+from great_expectations.types.attributes import Attributes
+from great_expectations.rule_based_profiler.parameter_container import (
+    ParameterContainer,
+    FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY,
+    FULLY_QUALIFIED_PARAMETER_NAME_ATTRIBUTED_VALUE_KEY,
+    FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY,
+)
+
+
+class DummyDomain(Domain):
+    """
+    To set up execution of "ParameterBuilder.build_parameters()" only "id" property of "Domain" is required.
+    """
+
+    def __init__(self, domain_type=MetricDomainTypes.TABLE):
+        super().__init__(domain_type=domain_type)
+
+    @property
+    def id(
+        self,
+    ) -> str:
+        return "my_id"
+
+
+class DummyParameterBuilder(ParameterBuilder):
+    """
+    Since goal of these tests is to ensure that private implementation method, "ParameterBuilder._build_parameters()",
+    of public interface method, "ParameterBuilder.build_parameters()", is called (as proof that
+    "evaluation_parameter_builder" dependencies of given "ParameterBuilder" are executed), all production functionality
+    is "mocked", and "call_count" property is introduced and incremented in relevant method for assertions in tests.
+    """
+
+    exclude_field_names: ClassVar[Set[str]] = ParameterBuilder.exclude_field_names | {
+        "call_count",
+    }
+
+    def __init__(
+        self,
+        name: str,
+        evaluation_parameter_builder_configs: Optional[
+            List[ParameterBuilderConfig]
+        ] = None,
+    ) -> None:
+        super().__init__(
+            name=name,
+            evaluation_parameter_builder_configs=evaluation_parameter_builder_configs,
+            data_context=None,
+        )
+
+        self._call_count: int = 0
+
+    @property
+    def call_count(self) -> int:
+        return self._call_count
+
+    def _build_parameters(
+        self,
+        domain: Domain,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+        runtime_configuration: Optional[dict] = None,
+    ) -> Attributes:
+        """
+        Only "_call_count" is incremented; otherwise, arbitrary test value is returned.
+        """
+        self._call_count += 1
+
+        return Attributes(
+            {
+                FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY: 1,
+                FULLY_QUALIFIED_PARAMETER_NAME_ATTRIBUTED_VALUE_KEY: {"batch_id_0": 1},
+                FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY: {"environment": "test"},
+            }
+        )
+
```

Comment:
```suggestion
```

Reply:
Instead of this, would patch object on function `__build_parameters` for the specific ParameterBuilder used in test cases work?

ie `with patch.object(ProductionClass, 'method', return_value=Attributes(...)) as mock_method:` 
documentation: https://docs.python.org/3/library/unittest.mock.html

Reply:
@TrangPham About half a year ago, the team has adopted the approach of actually creating fake and dummy objects.  Also, the "patch.object" approach has a nicer form (without the "object").  I am actually using its method decorator form (they are equivalent, and in the current situation, this "with" would encapsulate most of the logic of the method, so I just put it at the top for cleanliness).  Happy to discuss.  Thank you.